### ISSUE TITLE
Creating a Workbook with a Pivot Table in Microsoft Excel and saving with EPPlus results in Excel crashing when trying to open that file.  

### ISSUE THREAD

**Issue by art-ist** [JanKallman/EPPlus#99]
## Issue

I'm trying to use an Excel file as a template for reporting that can be modified by a user. After the template workbook containing a data table and a pivot table is generated, I just want to open the template, change the data and save the new file. That works well, as long as the template does not contain a pivot table. If the template contains a pivot table Excel crashes when trying to open the resulting xlsx file.

## Steps to reproduce

1. Create an Excel Workbook using Excel 2016 (see attached Template.xlsx)
    1. add a simple table with data
    2. add a pivot table analyzing that table
2. open (optionally modify) and save the file using EPPlus (see code sample and Results.xlsx)

```c#
using System.IO;
using OfficeOpenXml;

namespace EPPlusTest {
    class Program {
        static void Main(string[] args) {

            var template = "./Template.xlsx";
            var result = "./Result.xlsx";
            using (var inStream = File.Open(template, FileMode.Open)) {
                using (var outStream = File.Open(result, FileMode.OpenOrCreate, FileAccess.ReadWrite)) {
                    using (ExcelPackage xl = new ExcelPackage(outStream, inStream)) {
                        xl.Save();
                    }
                }
            }

        }
    }
}
```
(Test project including the Template.xlsx and Result.xlsx included. [EPPlusTest.zip](https://github.com/JanKallman/EPPlus/files/1629471/EPPlusTest.zip) )

## Notes

#25 sounds quite similar. But as copying my pivot- and data worksheets to the samlpe12.xlsx file, mentioned there doesen't solve the problem, I'm not sure if it's the exact same thing. For me it looks like EPPlus somehow corrupts a pivot table that was created by Excel.

When I open the Result.xlsx in LibreOffice Calc te file opens and the pivot table works just fine only the data worksheet shows non working filter dropdown-buttons in each row after the last data row.

**Comment by art-ist** [comment]
## Workaround
Some more digging: I found that:

- creating the template (Data and PivotTable) using EPPlus (like it's shown in the sample12 of this project) works without problems. 
  ```cs
         using (ExcelPackage xl = new ExcelPackage(outputStream)) {
            //DataTable dt = ... ;

            var wsResults = xl.Workbook.Worksheets.Add("Results");
            var wsData = xl.Workbook.Worksheets.Add("Data");
            
            var dataCell = wsData.Cells["A1"];
            var dataRange = dataCell.LoadFromDataTable(dt, PrintHeaders: true);
            var pivot = wsResults.PivotTables.Add(wsResults.Cells["A1"], dataRange, "Results");

            xl.Save();
        }
  ```
- Microsoft Excel then can be used to open and modify the template workbook.
- Then EPPlus can be used to create new workbooks based on that template replacing the data.
  ```cs
        using (OfficeOpenXml.ExcelPackage xl = new OfficeOpenXml.ExcelPackage(outputStream, templateStream)) {
            //DataTable dt = ... ;

            var dataCell = wsData.Cells["A1"];
            var dataRange = dataCell.LoadFromDataTable(dt, PrintHeaders: true);

            //find PivotTables using data
            foreach (var sheets in xl.Workbook.Worksheets) {
                foreach (var pivot in sheets.PivotTables.Where(p => p.CacheDefinition.SourceRange.Start.Address == dataCell.Start.Address)) {
                    pivot.CacheDefinition.SourceRange = dataRange;
                }
            }
            xl.Save();
        }
  ```
- This time the resulting document opens just fine in Microsoft Excel.

## It's still an issue
Though this takes out most of the pain of my current use case. Still, just piping a working Excel workbook through EPPlus leading to a document that crashes Microsoft Excel still is a relevant bug. 

By the behavior observed, I assume it's actually the very same problem as issue #25.

**Comment by JanKallman** [comment]
Thanks for the detailed description. I tried the attached code/template with the latest source and it seems to work. Issue #25 should have been fixed, so I assume you use EPPlus 4.1.x? Can you try using the latest beta version (4.5.0.1-Beta) to verify that it solves this issue?

**Comment by art-ist** [comment]
Yeah, **I can confirm, the issue is solved** in the current beta. Great!
(I overlooked, that the Nuget package is still on 4.1. Do you already have a time frame for the release of 4.5?)

**Comment by ac931274** [comment]
Hello, I also was having this issue and the latest beta resolves it.  Thanks

**Comment by JanKallman** [comment]
Fixed in 4.5.1
